include/translatable-objects.php

```diff
@@ -0,0 +1,121 @@
+<?php
+/**
+ * @package Polylang
+ */
+
+/**
+ * Registry for all translatable objects.
+ *
+ * @since 3.4
+ *
+ * @phpstan-implements IteratorAggregate<non-empty-string, PLL_Translatable_Object>
+ */
+class PLL_Translatable_Objects implements IteratorAggregate {
+
+	/**
+	 * Type of the main translatable object.
+	 *
+	 * @var string
+	 *
+	 * @phpstan-var non-empty-string|null
+	 */
+	private $main_type;
```

Comment:
I suggest not changing the type when we can.
```suggestion
	 */
	private $main_type = '';
```

Reply:
So $main_type can't be non-empty-string 😁